### PR TITLE
Improve performance of the Vite integration

### PR DESCRIPTION
**Pull request by caendesilva**
### Abstract

Follow up to https://github.com/hydephp/develop/pull/2016, to fix Windows performance with a much faster "hotfile" system like [Laravel uses](https://github.com/laravel/vite-plugin/blob/1.x/src/index.ts), instead of a slow socket connection test.

- Targets https://github.com/hydephp/develop/pull/1565 via https://github.com/hydephp/develop/pull/2006

### Details

This pull request includes changes to improve the performance and functionality of the Vite development server integration. The most important changes include replacing the Vite dev server check with a hot file check and restructuring the Vite configuration to manage the hot file lifecycle.

Improvements to Vite dev server integration:

* [`packages/framework/src/Facades/Vite.php`](diffhunk://#diff-e742d53af81236a3ecf137e0060198510327715e1054f249f03056b5890cb3e8L21-R22): Replaced the Vite dev server check with a more efficient hot file check to improve performance, especially on Windows.

Restructuring Vite configuration:

* [`vite.config.js`](diffhunk://#diff-58e6f63d87181b1c6a8cb6e5f1691df04aa32854456efcd52ca71c8541375d26L12-R26): Introduced a new `hydeVitePlugin` to handle the creation and removal of the Vite hot file when the server starts and stops, respectively. This ensures the hot file accurately reflects the server's state. [[1]](diffhunk://#diff-58e6f63d87181b1c6a8cb6e5f1691df04aa32854456efcd52ca71c8541375d26L12-R26) [[2]](diffhunk://#diff-58e6f63d87181b1c6a8cb6e5f1691df04aa32854456efcd52ca71c8541375d26R37-R49)

### Benchmark

Instead of the the ~100ms delay on Windows when running `npm run dev` manually instead of through `php hyde serve --vite` we now only have a `TBD` delay for each page preview.

Starting/stopping the Vite server brings a tiny performance impact, which is negligible in this context:

```
Signal Handler Setup: 0.23ms

  VITE v5.4.10  ready in 840 ms

  ➜  Local:   http://127.0.0.1:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
Cleanup Operation: 0.981ms
```

### CODE REVIEW

File: vite.config.js [hunk 1]
```diff
@@ -9,6 +9,34 @@ import autoprefixer from 'autoprefixer';
...
+        // Create hot file when Vite server starts
+        fs.writeFileSync(path.resolve(process.cwd(), 'app/storage/framework/cache/vite.hot'), '');
+
+        // Remove hot file when Vite server closes
+        ['SIGINT', 'SIGTERM'].forEach(signal => {
+            process.on(signal, () => {
+                fs.rmSync(path.resolve(process.cwd(), 'app/storage/framework/cache/vite.hot'));
+                process.exit();
+            });
+        });
```
Comment (caendesilva) [inline vite.config.js:24]: ### Options Tried

1. **Vite Plugin Lifecycle Hooks**:
   - **`closeBundle`**: Intended to clean up after the build process, but it didn't trigger reliably in this case.
   - **`buildEnd`**: Meant to handle cleanup after the build, but similarly, it didn't work as expected.
   - **Server Events**: Attempted to use `server.httpServer.on('close')` and `server.watcher.on('close')`, but these events also failed to trigger consistently during development.

2. **Process Signal Listeners**:
   - **`SIGINT`**: Captured when the user manually stops the server (e.g., using Ctrl+C). This worked reliably and allowed for cleanup of the `vite.hot` file.
   - **`SIGTERM`**: Sent by process managers or during system shutdown. Included for completeness.

### Rationale for Using Node.js Signal Listeners

- **Reliability**: Signal listeners for `SIGINT` and `SIGTERM` provide a dependable way to handle cleanup operations. They ensure that applications can respond to termination requests, allowing for graceful shutdowns.
  
- **Simplicity**: Using process signals simplifies the code by focusing on a straightforward mechanism for cleanup, rather than relying on various Vite lifecycle hooks and server events that may not trigger consistently.

- **Common Practice**: Handling process signals is a standard practice in Node.js applications, especially for server-side code. It ensures that resources are released properly, preventing potential issues like file locks or incomplete operations.

- **Flexibility**: By listening for both `SIGINT` and `SIGTERM`, the application can handle different shutdown scenarios, making it more robust in various environments (development, production, containerized setups).

In conclusion, while various options were explored, the use of Node.js signal listeners simed to be the most effective and reliable method for managing cleanup tasks in the Vite plugin.
